### DISCUSSION TITLE
Game Board

### DISCUSSION THREAD

**Discussion by KenWillmott** [KenWillmott/HD6303EVB#13] [General]
Here is the almost finished game expansion board, TMS9918A VDP and Philips sound generator, 8 bit dual joystick input interface:
![image](https://github.com/KenWillmott/HD6303EVB/assets/17345651/6e6f0ad9-d1d4-4d87-8832-12198c86f28d)
the memory array looks like this:
![image](https://github.com/KenWillmott/HD6303EVB/assets/17345651/bf704389-efa4-42c1-9f18-364f75c57485)

**Comment by TheEccentricGenius** [comment]
I can see why it's taken so long....Very nice (not that I am a composite video expert).

The memory has expanded since I last saw....Or was that a different board?

**Comment by KenWillmott** [comment]
The VDP uses 16k*8 dynamic RAM, which it has exclusive access to. I spent 2 weeks trying to find a way to emulate it with static RAM or 4 bit wide DRAMs, but the late write timing makes it difficult or impossible. At least, they are only 16 pin IC's.

**Comment by KenWillmott** [comment]
It's almost ready for fabrication, and I managed to add a serial port. It's on a 10cm x 10cm board because it seems that I can save some money by not exceeding those dimensions:
![image](https://github.com/KenWillmott/HD6303EVB/assets/17345651/a7ea5377-22f4-42c9-bea6-4cf9a9106f9a)
I think, the next processor board I make will have the same dimensions, mounting holes, and bus connector location. I've also made it card cage compatible, leaving a bit of keep out area on the sides, and placing all the I/O on the opposite side from the bus connector. So, it can be inserted in a cage with backplane and card guides, or else mounted as a mezzanine board (stacked) on top of a processor board using the M3 mounting holes and standoffs.

**Reply by TheEccentricGenius** [reply]
All the things I would forget till I had one and tried it.

**Comment by KenWillmott** [comment]
I did forget a lot of those things on the EVB, and having an actual board on my desk did certainly make it easier to notice! A run of 20 game board PCBs is finished fabrication and in shipment right now. It will be exciting to try one out. In fact, I find that it is nearly impossible to look ahead to a final system design without trying out some things. I did a survey of some different legacy boards and systems from the era, but the development environment was different back then. The costs were a lot higher, memory was a lot less dense, and design was a kind of "wild west" with companies trying a lot of different things, often focused mainly on cost. We can now build a lot more functionality at reasonable cost with a lot of improvements like size, power consumption etc., which means really a more "professional" system is easy to achieve.

We can learn from past mistakes. The Apple One had a timing problem because Wozniak used phase 1 instead of phase 2 for some timing. At least one peripheral board had a one-shot delay circuit kludge to fix that. We want to get things 100% right from the start.

On another subject, notice that the ACIA (UART) on the game board provides a serial interface for processors that don't have one built in. This will be very helpful for any 6809, 6502 or any processor board that doesn't have room for it, or where two serial ports are desired.

**Comment by KenWillmott** [comment]
Board is finished, but I can't test it until I get the 7.3728MHz oscillator modules - the one in the photo is a placeholder for the picture, it's the wrong value.
![image](https://github.com/KenWillmott/HD6303EVB/assets/17345651/99e14546-7337-4443-81bb-9dc0cafa65ba)

**Reply by TheEccentricGenius** [reply]
Excellent job, I would be very pleased to have a board like that.
Is it a 6-layer board?

**Comment by KenWillmott** [comment]
As a cost saving, it is a 2 layer board. I'm going to open a new repository for it this week, then you can have a closer look. You could have a board now gratis (any interested developer could actually), but as you know the ways of getting it around the world aren't cheap.

**Reply by TheEccentricGenius** [reply]
OK. I'll look out for the repository.

**Reply by TheEccentricGenius** [reply]
I finally saw the repo....I'll get the HD6303RP up to date tomorrow hopefully.

**Comment by KenWillmott** [comment]
Full testing is held back because I don't have the 7.3728 oscillator cans yet. So I'm looking forward to testing the synthesizer and ACIA this week. I admit, coding in assembly vs. C++ does slow me down, too!
